django/core/management/commands/migrate.py

```diff
@@ -38,6 +38,9 @@ def add_arguments(self, parser):
                 'Defaults to the "default" database.')
         parser.add_argument('--fake', action='store_true', dest='fake', default=False,
             help='Mark migrations as run without actually running them')
+        parser.add_argument('--fake-initial', action='store_true', dest='fake_initial', default=False,
+            help='Detect if an initial migration has been applied before and fake it. Make sure '
```

Comment:
In the usual case for using this, it wouldn't be because "an initial migration has been applied before,", it'd be because the database pre-existed any (Django) migrations at all. 

Also, it's really the contents of your initial migration file that you need to compare to, not your model definitions.

Suggested wording: "Detect if tables already exist and fake-apply initial migrations if so. Make sure that the current database schema matches your initial migration before using this flag!"

As a bonus, this also hints at the fact that the automated check here is no more sophisticated than just checking if tables exist.